### ISSUE TITLE
Support do (x) ->

### ISSUE THREAD

**Issue by TrevorBurnham** [satyr/coco#33]
Currently, `do (x = y) -> ...` compiles to

```
(function(x){
  x == null && (x = y);
  ...
}());
```

which is really just an inefficient way of writing

```
(function(x){
  ...
}(y));
```

Changing `do` to rewrite in this way would also allow you to write `do (x = x)`, which currently is nonsensical, to capture a particular value of `x` (useful in loops).

I would also suggest that `do (x)` be made a shorthand for `do (x = x)`.

**Comment by satyr** [comment]
You sure that it's _satyr/coco_, rather than _jashkenas/coffee-script_? ;)

> `(function(x){}(y))`

For this usage, you can simply `do -> x = y`. This `x` is guaranteed to be closed, as Coco doesn't implicitly share scopes like Coffee.

Note that Coco also implements an aid for the loop variables issue (see **scoped** section in [additions](https://github.com/satyr/coco/wiki/additions)), DRYing you up.

**Comment by satyr** [comment]
closed by 4e7b794c1121b78243fd88c73b543c29503b321a; made `do` Coffee 1.0.0 compliant; `do (x = y, z) -&gt;` -&gt; `(function(x, z){}(y, z));`

**Comment by satyr** [comment]
Wontfix-ed as per #42.
